tests/integration/test_update.py

```diff
@@ -80,25 +134,56 @@ def mock_download_remote_sources():
 
     flexmock(SpecFile, download_remote_sources=mock_download_remote_sources)
 
-    flexmock(DistGit, push_to_fork=lambda branch_name: None)
+    flexmock(DistGit, push_to_fork=lambda *args, **kwargs: None)
 
     def mocked_new_sources(sources=None):
         if not Path(sources).is_file():
             raise RuntimeError("archive does not exist")
     flexmock(FedPKG, init_ticket=lambda x=None: None, new_sources=mocked_new_sources)
 
+    pc = get_local_package_config(str(u))
+    pc.downstream_project_url = str(d)
+    pc.upstream_project_url = str(u)
+    # https://stackoverflow.com/questions/45580215/using-flexmock-on-python-modules
+    flexmock(sys.modules["packit.bot_api"]).should_receive("get_packit_config_from_repo").and_return(pc)
+    return u, d
+
 
-@pytest.mark.xfail
 def test_basic_local_update(upstream_n_distgit, mock_update_workflow):
     """ basic propose-update test: mock remote API, use local upstream and dist-git """
     u, d = upstream_n_distgit
 
     chdir(u)
-    c = Config()
-    pc = PackageConfig()
+    c = get_test_config()
+
+    pc = get_local_package_config(str(u))
+    pc.upstream_project_url = str(u)
+    pc.downstream_project_url = str(d)
     api = PackitAPI(c, pc)
-    api.sync_pr()
+    api.sync_release("master", "0.1.0")
+
+    assert (d / TARBALL_NAME).is_file()
+    spec = SpecFile(str(d / "beer.spec"), None)
+    assert spec.get_full_version() == "0.1.0"
+
 
+def test_single_message(github_release_fedmsg, mock_update_workflow):
+    u, d = mock_update_workflow
+
+    conf = get_test_config()
+    api = PackitBotAPI(conf)
+    api.sync_upstream_release_with_fedmsg(github_release_fedmsg)
     assert (d / TARBALL_NAME).is_file()
     spec = SpecFile(str(d / "beer.spec"), None)
     assert spec.get_full_version() == "0.1.0"
+
+
+def test_loop(mock_update_workflow, github_release_fedmsg):
+    def mocked_iter_releases():
+        msg = copy.deepcopy(github_release_fedmsg)
+        yield msg["topic"], msg
+        return
```

Comment:
is the `return` needed?

Reply:
dunno, wasn't sure if you need return when you have yield; me no python guru, leave me alone